### ISSUE TITLE
Possible crash in Orchard.Indexing | Orchard Concurrency Issues

### ISSUE THREAD

**Issue by Xceno** [OrchardCMS/Orchard#7937]
## TLDR;
CC: @BuzzKeith @Hazzamanic 

If (in whatever way) a `ContentItemRecord` with a `ContenType_Id` of `null` gets saved to `Orchard_Framwork_ContentItemRecord`. Then [this code](https://github.com/OrchardCMS/Orchard/blob/1.10.x/src/Orchard.Web/Modules/Orchard.Indexing/Services/IndexingTaskExecutor.cs#L163) and [this code](https://github.com/OrchardCMS/Orchard/blob/1.10.x/src/Orchard.Web/Modules/Orchard.Indexing/Services/IndexingTaskExecutor.cs#L224) in `Orchard.Indexing` will crash with a `NullReferenceException`.

This will prevent the recreation and any updates of Lucene indices.
(Noticeable by an error log that grows every few seconds and a DocumentCount that will always remain zero)

I'm going to provide a PR for that shortly. It will filter out corrupted records so that the indexing task will continue to work, even if those records remain unfixed inside the DB.

## The long version
For some reason that wasn't exactly pinned down yet, Orchard seems to have some concurrency problems that lead to corrupted datasets, invalid validations, or something along those lines.
This was probably uncovered the first time by @HermesSbicego-Laser and @MatteoPiovanelli-Laser in issue #7708 while testing Orchard with their [custom scripts](https://github.com/LaserSrl/Newman-Orchard-Parallel-Tests-).

Through those scripts I could confirm some issues with Autoroute, namely duplicated Pages and duplicated URLs, some days later. I was getting those issues since at least Orchard 1.5 or even sooner but never could reliably reproduce them until this day. This seems to be tracked now in #7720.

Fast forward to 2018:
After asking in the Gitter chat if someone has seen similar issues with Orchard.Indexing, [we came to the assumption](https://gitter.im/OrchardCMS/Orchard/archives/2018/01/11) that the issue described above also stems from some sort of concurrency problem, which yet needs to be tracked down.
